File: 3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        int n = nums.size();
        vector<int> arr(n, -1);
        for (int i = 0; i < n; i++) {
            int x = nums[i];
            for (int j = 0; j <= x; j++) {
                if ((j | (j + 1)) == x) {
                    arr[i] = j;
                    break;
                }
            }
        }
        return arr;
    }
};

File: LeetCode/3314-construct-the-minimum-bitwise-array-i/3314-construct-the-minimum-bitwise-array-i.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        arr = [-1] * n
        for i in range(n):
            x = nums[i]
            for j in range(0, x+1):
                if (j | (j + 1)) == x:
                    arr[i] = j
                    break
        return arr